src/mame/drivers/tetrisp2.cpp

```diff
@@ -1860,53 +1839,174 @@ TIMER_DEVICE_CALLBACK_MEMBER(stepstag_state::field_cb)
 	// irq 4 is definitely a 30 Hz-ish here as well,
 	// except we have a multi-screen arrangement setup and no way to pinpoint source
 	m_subcpu->set_input_line(4, HOLD_LINE);
+	stepstag_state::simulate_pc();	// any better place to put it into schedule?
 }
 
 void stepstag_state::setup_non_sysctrl_screen(machine_config &config, screen_device *screen, const XTAL xtal)
 {
 	// TODO: unknown clock and parameters
-	// assume there's a 42.954 MHz/6 like nndmseal to compensate the higher res
-	screen->set_raw(xtal/6, 455, 0, 352, 262, 0, 240);
+	screen->set_raw(xtal, 429, 0, 352, 525, 0, 240);	// should be 0-479 interlaced
 }
 
+void stepstag_state::simulate_pc()
+{
+	/*command list(stepstag):
+	$010x - start mpeg playing?
+	$02xx - ?
+	$03xx - open mpeg file set
+		$0311 jaleco logo animation
+		$0319 demo L/R bg
+	$04xx - ?
+	$07xx - load sound files to particular addresses of ymz280b ram
+		$070x stereo game music
+		$071x sound effects
+		$072x sample set for music selection
+		$07ff clear music sample region?
+	*/
+	FILE *inFile;
+	unsigned int size, position;	// position2
+	const unsigned int total_ram = 0x1000000;	// full 16MB for each ymz280b
+	if(strcmp(machine().system().name,"vjdash") == 0)
+		position = 0x400000;
+	//	position2 = 0xa00000;	// bass?
+	else
+		position = 0x900000;
+
+	if((m_stepstag_main2pc & 0xff0) == 0x700)
+	{
+		inFile = fopen("bgm_l", "rb");
```

Comment:
1. These files should really be tied into the ROM structure instead and marked as BAD_DUMP. Would be slightly better to partition them into an adhoc CHD but that may come up later.
2. as an aside of above: that also means that whatever reading should honor reading from the memory region itself and use `std::copy_n`, making this way easier to follow up.